### DISCUSSION TITLE
Plotting an xarray of size (nx, ny) in 2D whatever nx and ny

### DISCUSSION THREAD

**Discussion by patquem** [pydata/xarray#4813] [Q&A]
Hello,

I work with 2D arrays of variable sizes and I wish to represent them (always) in 2D representation.
Unfortunately, when one of the sizes is set to 1, xarray.plot() automatically degenerates the plotting toward a 1D signal representation.
Is there a way to prevent this ? (like an hidden option to force the 2D representation in any circunstances ?).
Thanks,
Patrick

```
import numpy as np
import xarray as xr
import matplotlib.pyplot as plt

nx = 2 # OK
# nx = 1 # NOK -> 1D signal
arr = np.random.random((nx, 10))
x = np.arange(nx)
y = np.arange(10)
xarr = xr.DataArray(arr, coords=[x, y])
xarr.plot()
plt.show()
```

**Comment by dcherian** [comment]
See #620

**Reply by patquem** [reply]
Thank you dcherian for your (very fast) answer
Indeed the post #620 is similar to my request. But I don't see what it has been done finally to respect the request ?
I can't believe it 's still opened since 2015 ? @jhamman ?

**Comment by johnomotani** [comment]
A simple workaround might be to call `imshow` explicitly, like
```
import numpy as np
import xarray as xr
import matplotlib.pyplot as plt

nx = 1
arr = np.random.random((nx, 10))
x = np.arange(nx)
y = np.arange(10)
xarr = xr.DataArray(arr, coords=[x, y])
xarr.plot.imshow()
plt.show()
```
`pcolormesh()` instead of `imshow()` doesn't work for me - shows a blank plot even with `infer_intervals=False`, not sure why...
